src/rules/ResourceSourceChecker.js

```diff
@@ -45,97 +46,38 @@ class ResourceSourceChecker extends Rule {
      */
     constructor(options) {
         super(options);
-
-        if (!options || !options.name || !options.description || !options.note || !options.regexps) {
-            throw "Missing required options for the ResourceMatcher constructor";
-        }
-        ["name", "description", "regexps", "note", "sourceLocale", "link"].forEach(prop => {
-            this[prop] = options[prop];
-        });
-        this.sourceLocale = this.sourceLocale || "en-US";
-
-        // this may throw if you got to the regexp syntax wrong:
-        this.re = this.regexps.map(regexp => new RegExp(regexp, "gu"));
-    }
-
-    getRuleType() {
-        return "resource";
     }
 
     /**
      * @override
      */
-    match(options) {
-        const { locale, resource, file } = options || {};
-        const _this = this;
-
-        /**
-         * @private
-         */
-        function checkString(re, src) {
-            re.lastIndex = 0;
-            let matches = [];
-            const strippedSrc = stripPlurals(src);
-
-            // check the target only
-            re.lastIndex = 0;
-            let match = re.exec(strippedSrc);
-            while (match) {
-                let value = {
-                    severity: "error",
-                    id: resource.getKey(),
-                    rule: _this,
-                    pathName: file,
-                    highlight: `Source: ${src.substring(0, match.index)}<e0>${match[0]}</e0>${src.substring(match.index+match[0].length)}`,
-                    description: _this.note.replace(/\{matchString\}/g, match[0])
-                };
-                if (typeof(options.lineNumber) !== 'undefined') {
-                    value.lineNumber = options.lineNumber;
-                }
-                matches.push(new Result(value));
-
-                match = re.exec(strippedSrc);
+    checkString(re, src, file, resource) {
+        re.lastIndex = 0;
+        let matches = [];
+        const strippedSrc = stripPlurals(src);
+
+        // check the target only
+        re.lastIndex = 0;
+        let match = re.exec(strippedSrc);
+        while (match) {
+            let value = {
+                severity: this.severity,
+                id: resource.getKey(),
+                rule: this,
+                pathName: file,
+                highlight: `Source: ${src.substring(0, match.index)}<e0>${match[0]}</e0>${src.substring(match.index+match[0].length)}`,
+                description: this.note.replace(/\{matchString\}/g, match[0])
```

Comment:
Missing locale in result (see comment under ResourceTargetChecker)

Reply:
same ^